src/main/java/yeelp/scalingfeast/handlers/EnchantmentHandler.java

```diff
@@ -91,6 +84,42 @@ public void onAttackEvent(LivingAttackEvent evt)
 		}
 	}
 	
+	@SubscribeEvent(priority=EventPriority.LOWEST)
+	public void getMaxExhaustion(ExhaustionEvent.GetMaxExhaustion evt)
+	{
+		int level = EnchantmentHelper.getMaxEnchantmentLevel(SFEnchantments.lazinessCurse, evt.player);
+		if(level != 0)
+		{
+			evt.maxExhaustionLevel /= 2.0f;
+		}
+	}
+	
+	@SubscribeEvent(priority=EventPriority.LOWEST)
+	public void getStarvationRate(StarvationEvent.GetStarveTickPeriod evt)
+	{
+		int level = EnchantmentHelper.getMaxEnchantmentLevel(SFEnchantments.deprivationCurse, evt.player);
+		if(level != 0)
+		{
+			evt.starveTickPeriod /= 2.0f;
+		}
+	}
+	
+	@SubscribeEvent
+	public void onFoodStatsAddition(FoodEvent.FoodStatsAddition evt)
+	{
+		EntityPlayer player = evt.player;
+		int level = EnchantmentHelper.getMaxEnchantmentLevel(SFEnchantments.sensitivityCurse, player);
+		if(level != 0 || ModConfig.items.enchants.globalSensitvity)
+		{
+			int overflow = (evt.foodValuesToBeAdded.hunger + player.getFoodStats().getFoodLevel()) - ScalingFeastAPI.accessor.getModifiedFoodCap(player);
+			ScalingFeast.debug(Integer.toString(overflow));
+			if(overflow > 0)
+			{
+				player.addPotionEffect(new PotionEffect(SFPotion.softstomach, 20*20, overflow-1));
```

Comment:
Assuming this is `new PotionEffect(name, duration, level)`, what happens if you somehow eat a food worth 256 food points or more (achievable with high enough Gluttony level) while you are 1 hunger down from your cap and the Curse of Sensitivity effect is active? Would the level overflow, or could it be changed to just cap the Soft Stomach level at 255?

Reply:
Internally Minecraft saves the potion amplifier as an ```int```, but in game I'm not quite sure what would happen, probably not something well defined. hard capping at 255 is good fail safe.